File: node_tutorials/hello_world_producer.js
//Dependencies
import { AMQPClient } from '@cloudamqp/amqp-client'
import {} from 'dotenv/config'

const cloudAMQPURL = process.env.CLOUDAMQP_URL

async function startProducer() {
  try {
    //Setup a connection to the RabbitMQ server
    const connection = new AMQPClient(cloudAMQPURL)
    await connection.connect()
    const channel = await connection.channel()

    console.log("[✅] Connection over channel established")

    const q = await channel.queue('hello_world')

    //Publish a message to the exchange
    async function sendToQueue(routingKey, body) {
      //amqp-client function expects: exchange, routingKey, message, options
      await q.publish('', { routingKey }, body)
      console.log("[📥] Message sent to queue", body)
    }
  
    //Send some messages to the queue
    sendToQueue("hello_world", "Hello World");
    sendToQueue("hello_world", "Hello World");
    sendToQueue("wrong_routing_key", "Hello World");
  
    setTimeout(() => {
      //Close the connection
      connection.close()
      console.log("[❎] Connection closed")
      process.exit(0)
    }, 500);
    } catch (error) {
      console.error(error)

      //Retry after 3 second
      setTimeout(() => {
        startProducer()
      }, 3000)
  }
}

startProducer()

File: ruby_tutorials/hello_world_producer.rb
require "amqp-client"
require "dotenv/load"

# Grab lavinmq url from .env file
lavinmq_url = ENV['CLOUDAMQP_URL']

# Opens and establishes a connection
connection = AMQP::Client.new(lavinmq_url).connect

# Open a channel
channel = connection.channel
puts "[✅] Connection over channel established"

# Create a queue
channel.queue_declare("hello_world")

def send_to_queue(channel, routing_key, body):
    # Publish function expects: body, exchange, routing_key in that order
    channel.basic_publish(
        body
        '',
        routing_key
    )
    puts "[📥] Message sent to queue - msg:  #{body}"
  
  # Publish messages
  send_to_queue channel, "hello_world", "Hello World - 1"
  send_to_queue channel, "hello_world", "Hello World - 2"
  send_to_queue channel, "wrong_routing_key", "Hello World - 3"
  
  begin
    connection.close
    puts "[❎] Connection closed"
  rescue => exception
    puts "Error: #{exception}"
  end